src/components/global/Playground/index.tsx

```diff
@@ -182,19 +185,77 @@ export default function Playground({
+              fill="#3178C6"
+            />
+          </svg>
+        );
+      case 'html':
+        return (
+          <svg width="12" height="12" viewBox="0 0 12 12" fill="none" xmlns="http://www.w3.org/2000/svg">
+            <path
+              d="M3.75 8.625L0.75 6L3.75 3.375"
+              stroke="#EB4700"
+              strokeWidth="1.5"
+              strokeLinecap="round"
+              strokeLinejoin="round"
+            />
+            <path
+              d="M8.25 8.625L11.25 6L8.25 3.375"
+              stroke="#EB4700"
+              strokeWidth="1.5"
+              strokeLinecap="round"
+              strokeLinejoin="round"
+            />
+          </svg>
+        );
+      case 'vue':
+        return (
+          <svg xmlns="http://www.w3.org/2000/svg" width="12" height="12" viewBox="0 0 512 512">
+            <polygon
+              fill="#33475A"
+              points="256 144.03 200.51 47.92 121.08 47.92 256 281.61 390.92 47.92 311.49 47.92 256 144.03"
+            />
+            <polygon
+              fill="#3FB27F"
+              points="409.4 47.92 256 313.61 102.6 47.92 15.74 47.92 256 464.08 496.26 47.92 409.4 47.92"
+            />
+          </svg>
+        );
+    }
+  }
+
   function renderCodeSnippets() {
-    if (codeSnippets[usageTarget]) {
-      if (typeof codeSnippets[usageTarget].$$typeof === 'symbol') {
+    if (code[usageTarget]) {
+      if (!hasOutputTargetOptions) {
         return codeSnippets[usageTarget];
       }
-      return Object.keys(codeSnippets[usageTarget]).map((fileName) => {
-        return (
-          <div key={fileName} id={getCodeSnippetId(fileName)}>
-            <h4>{fileName}</h4>
-            {codeSnippets[usageTarget][fileName]}
-          </div>
-        );
-      });
+      return (
+        <PlaygroundTabs className="playground__tabs">
+          {Object.keys(codeSnippets[usageTarget]).map((fileName) => (
+            <TabItem
+              className="playground__tab-item"
+              value={fileName}
+              label={fileName}
+              key={fileName}
+              {...{
+                icon: getFileIcon(fileName),
+              }}
```

Comment:
This is extending the docusaurus `TabItem`, which doesn't have `icon` specified on its available props. To avoid having to have a custom `PlaygroundTabItem` just for the sake of passing props (which it does do), I used this approach to get around the typing issues.

This allows us to only have to maintain the outermost `Tabs` implementation and benefit from any docusaurus improvements to the tab item.

Reply:
Aha! Sneaky 😆